rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -154,7 +154,7 @@ func NewLayer2Relayer(ctx context.Context, l2Client *ethclient.Client, db *gorm.
 	}
 
 	// chain_monitor client
-	if cfg.ChainMonitor.Enabled {
+	if serviceType == ServiceTypeL2RollupRelayer && cfg.ChainMonitor.Enabled {
```

Comment:
if not added this, will cause a nil pointer error without chain-monitor configs.